### PR TITLE
Socials app - Add devise

### PR DESCRIPTION
**Pull request by rudicarrilloypr**
### Project requirements
- Remove `current_user` method in `ApplicationController`, devise will provide us with one.
- [Install and setup devise](https://github.com/heartcombo/devise#getting-started).
  - Can register a new user.
  - User logs in with a combination of email and password.
  - Hashed password should be stored in the database.
  - Ask for confirmation of email.
  - Can reset password.
- [Modify the views in devise](https://github.com/heartcombo/devise#configuring-views) for registration and login to match [the wireframes from the Sneak Peek](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md#:~:text=For%20this%20project%20you%20will%20have%20full%20freedom%20in%20terms%20of%20visual%20design%20but%20you%20will%20need%20to%20keep%20the%20following%20wireframes%3A) and your styling.

### CODE REVIEW

**Review by AbayomiOlaoye — CHANGES_REQUESTED** [review]
## STATUS: CHANGE REQUIRED ♻️ 

Hi @rudicarrilloypr :wave:  

Good job with this milestone so far! What a brilliant attempt!!
There are just a few more issues to work on before moving on to the next project but you are almost there!

so_close

## Highlights
- Descriptive Pull Request message :heavy_check_mark: 
- Good use of Gitflow :heavy_check_mark: 
- Meaningful commit messages :heavy_check_mark: 
- No linter error :heavy_check_mark: 
- Descriptive ReadMe documentation :heavy_check_mark: 
- Devise was added correctly :heavy_check_mark: 

## Required Changes ♻️
- [ ] _Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

- Additionally but optionally, you might also choose to create some users, posts, etc. as seed so that they can be added to the database by running `rails db:seed` and have a robust UI to interact with.

image

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Reem-lab** [comment]
Hello @rudicarrilloypr did you see any comments from the previous code reviewer?

**Comment by rudicarrilloypr** [comment]
hi @Reem-lab no! anyone! that why I re uploeaded the submission

**Comment by Reem-lab** [comment]
Did you reach out to him? @rudicarrilloypr

**Comment by AbayomiOlaoye** [comment]
@rudicarrilloypr and @Reem-lab Sorry for the omission. I must have forgotten to save the comment.

**Comment by rudicarrilloypr** [comment]
yes! via slack @Reem-lab

**Comment by AbayomiOlaoye** [comment]
It's about the route on the login wireframe: the wireframe recommends accessing the login at `.../login` but it pops No Route. I had to navigate to `.../users/sign_in` before the login views could be displayed.

**Comment by rudicarrilloypr** [comment]
ok @AbayomiOlaoye @Reem-lab working on it, ill make the commmit right asap

**Comment by AbayomiOlaoye** [comment]
> yes! via slack @Reem-lab

About this, you could have said that you are just making the attempt now because I am just seeing a message sent at `1:54am`. I apologise for the omission though.

File: config/routes.rb [hunk 1]
```diff
@@ -1,4 +1,5 @@
 Rails.application.routes.draw do
+  devise_for :users
   root 'users#index'
   
   resources :users, only: [:index, :show] do
```
Comment (AbayomiOlaoye) [inline config/routes.rb]: - [x] It's about the route on the login wireframe: the wireframe recommends accessing the login at `.../login` but it pops No Route. I had to navigate to `.../users/sign_in` before the login views could be displayed.
- Ideally, until a user is signed in, he should have access to the user's route Kindly fix this to appropriately match the requirement - user story for ease of navigation and protection of the integrity of your user's info.

**Comment by AbayomiOlaoye** [comment]
Just adding that for the record...Please ignore if you have implemented it, @rudicarrilloypr

**Comment by rudicarrilloypr** [comment]
thanks! @AbayomiOlaoye I just implemented it! @Reem-lab

**Review by Reem-lab — APPROVED** [review]
# Hi @rudicarrilloypr an @Diegogagan2587,  👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights

- [ ]  Followed Gitflow 💯
- [ ]  Descriptive PR⭐
- [ ]  Kudos for working on setup the devise🤩
- [ ]  No Linters error ✅
- [ ] Good job adding the routes🙏🏻


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**